src/server/game/Entities/GameObject/GameObject.cpp

```diff
@@ -2227,6 +2235,27 @@ bool GameObject::IsLootAllowedFor(Player const* player) const
     return true;
 }
 
+GameObject* GameObject::GetLinkedTrap()
+{
+    uint32 linkedEntry = GetGOInfo()->GetLinkedGameObjectEntry();
+
+    if (!linkedEntry)
+        return nullptr;
+
+    GameObject* linkedTrap = nullptr;
+
+    if (m_linkedTrap.Empty())
```

Comment:
m_linkedTrap should be set when spawned... see Spell::EffectSummonObjectWild for example

Using a grid searcher can get two nearby gameobjects controlling the same linked go spawn for example